### PR TITLE
Remove deprecated `datetime` functions

### PR DESCRIPTION
**Pull request by hussein-awala**
closes: #1133

### CODE REVIEW

File: pyiceberg/table/__init__.py [hunk 1]
```diff
@@ -3915,7 +3915,7 @@ def snapshots(self) -> "pa.Table":
                 additional_properties = None
 
             snapshots.append({
-                "committed_at": datetime.utcfromtimestamp(snapshot.timestamp_ms / 1000.0),
+                "committed_at": datetime.fromtimestamp(snapshot.timestamp_ms / 1000.0, tz=timezone.utc),
```
Comment (kevinjqliu) [inline pyiceberg/table/__init__.py:3918]: https://docs.python.org/3/library/datetime.html#datetime.datetime.utcfromtimestamp

File: pyproject.toml [hunk 2]
```diff
@@ -615,8 +615,6 @@ filterwarnings = [
   "ignore:unclosed <socket.socket",
   # Remove this in a future release of PySpark.
   "ignore:distutils Version classes are deprecated. Use packaging.version instead.",
-  "ignore:datetime.datetime.utcnow\\(\\) is deprecated and scheduled for removal in a future version.",
```
Comment (kevinjqliu) [inline pyproject.toml:618]: huh, not sure where this went, theres no references in the codebase
https://github.com/search?q=repo%3Aapache%2Ficeberg-python%20utcnow&type=code

**Review by kevinjqliu — APPROVED** [review]
LGTM!

File: pyproject.toml (hunk 2 above)
Comment (hussein-awala) [reply pyproject.toml:618]: it might be used in one of the dependency packages, but we don't need to filter the warning in that case

**Review by Fokko — APPROVED** [review]
Thanks for cleaning this up @hussein-awala Could you resolve the conflicts? Thanks!

**Comment by hussein-awala** [comment]
> Thanks for cleaning this up @hussein-awala Could you resolve the conflicts? Thanks!

I did and looks like the usage of the deprecated method was removed by another PR, this one now just removes the useless warning.

**Comment by kevinjqliu** [comment]
https://github.com/search?q=repo%3Aapache%2Ficeberg-python%20utcfromtimestamp&type=code
there's a few instances of `utcfromtimestamp`

**Review by kevinjqliu — APPROVED** [review]
LGTM! Thanks for working on this. 

Can you update the PR to reflect that this is only removing 1 out of 2 functions? 
And open partially close #1133

Perhaps we can also link #1133 as a comment above the filter warning

**Comment by Fokko** [comment]
@kevinjqliu Let's move this forward for now. Thanks @hussein-awala for fixing this 👍

**Pull request merged**